Source/WebCore/rendering/style/StyleRareInheritedData.h

```diff
@@ -209,6 +210,8 @@ class StyleRareInheritedData : public RefCounted<StyleRareInheritedData> {
 
     ListStyleType listStyleType;
 
+    std::optional<class ScrollbarColor> scrollbarColor;
```

Comment:
We should check with the `dump-class-layout`, but I think the `optional` will make it align at 12 bytes (4 for each StyleColor).
Using a custom `Markable<ScrollbarColor> when any of the underlying color is invalid could save 4 bytes (might be worth writing it as a commit)

Reply:
That command seg faults when I run it on my machine.

Apologies if this is a silly question, but there's only two StyleColours so where would the other 4 bytes be used I assume for the class instance itself?

Also this property cannot exist in a state with an invalid style colour. Either both the track and thumb colour are valid or this will be std::nullopt representing the auto value.

Reply:
Optional takes 1 bit (and aligned on 4 bytes) which could be saved by using Markable (instead of having the `nullopt`, we would use the already existing invalid states of either of the StyleColor).

A comment should be enough, it could definitely be in a follow-up patch.

Reply:
I will add a comment above this line that says

`// Investigate using Markable here to save on memory`

I've also made a follow up bug https://bugs.webkit.org/show_bug.cgi?id=257788